### ISSUE TITLE
Exception on logout when prepare_on_login: true

### ISSUE THREAD

**Issue by koekaverna** [scheb/2fa#107]
**Bundle version**: 5.11.0
**Symfony version**: 5.3.7
**PHP version**: 7.4.16
**Using authenticators** (`enable_authenticator_manager: true`): YES

**Description**
When prepare on login enabled (`prepare_on_login: true`) it cause error on logout route. 
In project I use custom Authenticator that authenticate with cookie token
It happens because `TwoFactorProviderPreparationListener@onLogin` is triggered before logout 
and set token from `AuthenticationEvent` 
```php
$this->twoFactorToken = $token;
```
After that symfony `LogoutListener` set `$this->tokenStorage->setToken(null);` and make logout routine.
After that in `TwoFactorProviderPreparationListener@onKernelResponse` -> `TokenPreparationRecorder@isTwoFactorProviderPrepared` checks the token from `TokenStorage` that is `null`
```php
        $token = $this->tokenStorage->getToken();
        if (!($token instanceof TwoFactorTokenInterface)) {
            throw new \RuntimeException('The security token has to be an instance of TwoFactorTokenInterface.');
        }
```

I think Listener should not make preparation on logout route somehow.

**To Reproduce**

**Additional Context**

```
ErrorException: The security token has to be an instance of TwoFactorTokenInterface. in /var/www/html/vendor/scheb/2fa-bundle/Security/TwoFactor/Provider/TokenPreparationRecorder.php:31
```

**Comment by koekaverna** [comment]
I found that it isn't good idea to prepare onLogin because this event trigger every time when client make request with 2fa token in header.
Feel free to close this issue.
Anyway it seems code isn't consistent it used token from different sources in on method.

**Comment by scheb** [comment]
> Anyway it seems code isn't consistent it used token from different sources in on method.

I'll keep it open until I got time to check. I might change this implementation a bit.

**Comment by scheb** [comment]
I checked it. It's a rare edge case, though I believe this issue should be handled. So I'll keep the bug open until it's fixed.

**Comment by scheb** [comment]
Fixed in 5.13.0
